docs/samples/bentoml/README.md

```diff
@@ -0,0 +1,171 @@
+management and model deployment functionality, giving ML teams an end-to-end model serving
+workflow, with DevOps best practices baked in.
+
+In this guide, we will demo serving a scikit-learn based iris classifier model with BentoML
+as InferenceService on a Kubernetes cluster with KFServing installed. The same deployment
+steps is also applicable for models trained with other machine learning frameworks, see
+more BentoML examples [here](https://docs.bentoml.org/en/latest/examples.html).
+
+## Deploy a custom InferenceService
+
+### Setup
+
+* Your ~/.kube/config should point to a cluster with KFServing installed.
+* Your cluster's Istio Ingress gateway must be network accessible.
+* Docker and Docker hub must be properly configured on your local system
+* Python 3.6 or above
+  * Install required packages `bentoml` and `scikit-learn` on your local system:
+
+    ```shell
+    pip install bentoml scikit-learn
+    ```
+
+### Build API model server using BentoML
+
+The followwing code defines an API model server that requires a `scikit-learn` model, and
+asks BentoML to figure out the required PyPI pip packages automatically. It also defined
+an API, which is the entry point for accessing this prediction service. And the API is expecting a `pandas.DataFrame` object as its input data.
+
+```python
+# iris_classifier.py
+from bentoml import env, artifacts, api, BentoService
+from bentoml.handlers import DataframeHandler
+from bentoml.artifact import SklearnModelArtifact
+
+@env(auto_pip_dependencies=True)
+@artifacts([SklearnModelArtifact('model')])
+class IrisClassifier(BentoService):
+
+    @api(DataframeHandler)
+    def predict(self, df):
+        return self.artifacts.model.predict(df)
+```
+
+The following code to create a BentoService SavedBundle with the trained an iris
+classification model. A BentoService SavedBundle is a versioned file archive ready for
+production deployment. The archive contains the model server defined above, python code
+dependencies, PyPi dependencies, and the trained iris classification model:
+
+```python
+# main.py
+from sklearn import svm
+from sklearn import datasets
+
+from iris_classifier import IrisClassifier
+
+if __name__ == "__main__":
+    # Load training data
+    iris = datasets.load_iris()
+    X, y = iris.data, iris.target
+
+    # Model Training
+    clf = svm.SVC(gamma='scale')
+    clf.fit(X, y)
+
+    # Create a iris classifier service instance
+    iris_classifier_service = IrisClassifier()
+
+    # Pack the newly trained model artifact
+    iris_classifier_service.pack('model', clf)
+
+    # Save the prediction service to disk for model serving
+    saved_path = iris_classifier_service.save()
+```
+
+The sample code above can be found in the BentoML repository, run the code with the
+following command:
+
+```bash
+git clone git@github.com:bentoml/BentoML.git
+python ./bentoml/guides/quick-start/main.py
+```
+
+### Deploy InferenceService
+
```

Comment:
BentoML provides a convenient way of containerizing the model API server with Docker.  To create a docker container image for the sample model above:

Find the file directory ...